U Bay Backend Connect

Baozzz04 opened the pull request:
Connected the back-end to front-end
Changes:

1. Add server folder for back-end
2. Change front-end code to fetch json from back-end using axios and cors

sophia-hubscher commented on server/routes/cart.js:
```diff
@@ -0,0 +1,90 @@
+    }
+})
+
+// GET USER CART
```
Should we wait on cart functionality until we decide as a team that that's what we're going for? It wasn't in our mockups at all, so I don't know if we should necessarily go ahead on all this just yet (I think the cart idea is super cool, I just want to make sure we have like sketches and stuff first so we all know what this actually entails)

Baozzz04 replied:
Yup, I comment it out in the server.js, so we are not using the cart.js

sophia-hubscher commented on server/models/Product.js:
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose")
+
+const ProductSchema = new mongoose.Schema(
```
As I understand it, the schema should be:
id
title
category
description
condition
price
imageURLs (list)
postDate

I wonder if we might also want to add a customerID/username field for the customer who posted it

Baozzz04 replied:
We can add customerID if we want later on. I think that it can also be related to our login ID

sophia-hubscher replied:
True! Sounds good

sophia-hubscher commented on server/models/User.js:
```diff
@@ -0,0 +1,13 @@
+const mongoose = require("mongoose")
+
+const UserSchema = new mongoose.Schema(
```
Alternatively, we could list productIDs here instead of customerIDs/email/whatever on the product schema. We could also do both, but redundancy isn't always great. What do you think?

Baozzz04 replied:
This part is the login process, which is related to what Sonny is doing. I don't want to interfere with what he has been doing, so I might as well delete/comment on this file. If we want to use it, I can merge it back in.

sophia-hubscher commented on src/App.js:
```diff
@@ -244,23 +265,27 @@ const App = () => {
+                          style={{ cursor: 'pointer' }}
+                        >
+                          <Paper style={{ display: 'flex', alignItems: 'center', padding: '10px', marginBottom: '10px' }}>
+                            <img src={"https://via.placeholder.com/150"} alt={listing.title} style={{ marginRight: '10px' }} />
```
We probably want to replace this placeholder URL with the images you're bringing in now that Griffin added those to the database

Baozzz04 replied:
Yes, I resolved the conflict and kept your previous code.

sophia-hubscher commented on src/App.js:
```diff
@@ -25,31 +27,47 @@ const theme = createTheme({
-const categories = ['Books', 'Furniture', 'Electronics', 'Clothing', 'Other'];
-const conditions = ['New', 'Like New', 'Very Good', 'Good', 'Acceptable'];
+const categories = ['Men', 'Women', 'Electronics', 'Clothing', 'Other'];
+const conditions = ['New', 'Like New', 'Very Good', 'Good', 'Acceptable', 'Old', 'Normal'];
```
This stuff is gonna be a little bit of a headache to merge—the updated App.js with all of Ashley's refactoring uses a util file to handle all of this stuff. We should be careful around this area when merging

Baozzz04 replied:
Well, I am not sure about this. Can someone help me with this merge request? It seems like the categories and conditions are not in the ListingLayout.js?

sophia-hubscher replied:
It might make the most sense for you and Ashley to work through this merge given she made most of the changes. If you look at the main branch though, the src/util.js file has:

```
import { createTheme } from '@mui/material/styles';

export const CATEGORIES = ['Books', 'Furniture', 'Electronics', 'Clothing', 'Other'];
export const CONDITIONS = ['New', 'Like New', 'Very Good', 'Good', 'Acceptable'];
export const SORTING = ['Best Match', 'Price: Low to High', 'Price: High to Low', 'Post Date: New to Old', 'Post Date: Old to New'];
export const MAXPRICE = 500;

export const pageTheme = createTheme({
    palette: {
      primary: {
        main: '#b2102f',
      },
      secondary: {
        main: '#6f252c',
      },
    },
    components: {
      MuiPaper: {
        styleOverrides: {
          root: {
            padding: '15px',
            gap: '5px',
            boxShadow: '0px 3px 5px rgba(0, 0, 0, 0.2)',
          },
        },
      },
    },
  });
```

so we just import with

```
import { CATEGORIES, CONDITIONS, SORTING, MAXPRICE } from './util';
```

in HomePage.js (the new App.js page)

Baozzz04 commented on server/models/Cart.js:
```diff
@@ -0,0 +1,16 @@
+const mongoose = require("mongoose")
+
+const CartSchema = new mongoose.Schema(
```
Sure, I will delete this file for now to keep it simple.

sophia-hubscher commented on src/Front-end/ListingsLayout.js:
```diff
@@ -1,235 +1,264 @@
-import React, { useState, useMemo } from 'react';
-import { Typography, Container, Grid, Paper, TextField, Button, ThemeProvider, List, ListItem, Checkbox, ListItemText, Slider, Input, MenuItem, Select } from '@mui/material';
-import { CATEGORIES, CONDITIONS, SORTING, MAXPRICE } from './util';
```
I think this is what's causing that error you posted on Slack! We should make sure we're using the stuff from util and the navigation bar from ./NavBar

Baozzz04 commented:
Overall changes for this branch:
1. Added server folder for back-end purposes
2. Connected the back-end to the front-end on the Listing page, so items from the database are displayed on the front page. 
3. Deleted some unrelated files in the server folder 
4. Changed some type code in the server model schema to send the image to front-end. Edited some front-end stuff to display the image on the listing page.
5. Wrote code for the sell page in front-end to post info to the database, and the listing page can pull that item to front page.

AshleyMcN commented:
Home Listings Page correctly fetches listings from the database.
